packages/safe-tools-client/app/components/future-payments-list/scheduled-payment-card/index.gts

```diff
@@ -41,21 +105,100 @@ export default class ScheduledPaymentCard extends Component<Signature> {
             {{!-- TODO: Add the right USD amount --}}
-            <span class="scheduled-payment-card__usd-amount">$ 1000 USD</span> 
+            <span class="scheduled-payment-card__usd-amount">$ 1000 USD</span>
           </div>
         </div>
       </div>
-      <BoxelIconButton
-        @icon="more-actions"
-        @height="30px"
-        aria-label="More Actions"
-      />
+
+      <BoxelDropdown>
+        <:trigger as |bindings|>
+          <BoxelIconButton
+            @icon="more-actions"
+            @height="30px"
+            {{bindings}}
+            data-test-scheduled-payment-card-options-button
+          />
+        </:trigger>
+        <:content as |dd|>
+          <BoxelMenu
+            @closeMenu={{dd.close}}
+            @items={{array
+              (menuItem
+                "Cancel Payment" (set this 'isCancelPaymentModalOpen' true)
+              )
+            }}
+          />
+        </:content>
+      </BoxelDropdown>
     </BoxelCardContainer>
+
+    <BoxelModal
+      @size='medium'
+      @isOpen={{this.isCancelPaymentModalOpen}}
+      @onClose={{noop}}
+      class="cancel-scheduled-payment"
+      data-test-cancel-scheduled-payment-modal
+    >
+      <BoxelActionContainer
+        as |Section ActionChin|
+      >
+        <Section @title="Cancel your scheduled payment">
+          <div>
+            <p>You're about to cancel your payment of <strong>{{weiToDecimal @scheduledPayment.amount this.tokenInfo.decimals}} {{this.tokenInfo.symbol}}</strong>
+            to <span class="blockchain-address">{{truncateMiddle @scheduledPayment.payeeAddress}}</span>, scheduled for <strong>{{formatDate @scheduledPayment.payAt "d/M/yyyy"}}</strong>.</p>
+
+            <p>This action will remove the scheduled payment from the scheduled payment module, and it won't be attempted in the future.</p>
+          </div>
+        </Section>
+
+        <ActionChin @state={{this.cancelPaymentState}}>
+          <:default as |a|>
+            {{#if this.cancelationErrorMessage}}
+              <a.ActionButton {{on "click" this.cancelScheduledPayment}} data-test-cancel-payment-button>
+                Cancel Payment
+              </a.ActionButton>
+
+              <a.CancelButton {{on 'click' this.closeCancelScheduledPaymentModal}} data-test-close-cancel-payment-modal>
+                Close
+              </a.CancelButton>
+
+              <a.InfoArea>
+                <FailureIcon class='action-chin-info-icon' />
+                <span>{{this.cancelationErrorMessage}}</span>
+              </a.InfoArea>
+            {{else if this.paymentCanceled}}
+              <a.ActionButton {{on "click" this.closeCancelScheduledPaymentModal}} data-test-close-cancel-payment-modal>
+                Close
+              </a.ActionButton>
+
+              <a.InfoArea>
+                <SuccessIcon class='action-chin-info-icon' />
```

Comment:
Since this seems like a common pattern, perhaps the InfoArea component should yield SuccessIcon and FailureIcon components with the class pre-applied. OK to do this in a separate PR.

Reply:
Yeah I agree. I added a ticket for that.